syntax = "proto2";

package tpcc_sql.validation.proto;

message Delivery {
  required uint32 w_id = 1;
  required uint32 d_id = 2;
  required uint32 o_carrier_id = 3;
  required uint32 ol_delivery_d = 4;
  required bool sequential = 5;
  repeated uint32 est_tables = 6;
}

message NewOrder {
  required uint32 w_id = 1;
  required uint32 d_id = 2;
  required uint32 c_id = 3;
  required uint32 ol_cnt = 4;
  required uint32 rbk = 5;
  repeated uint32 o_ol_i_ids = 6;
  repeated uint32 unique_items = 7;
  repeated uint32 o_ol_supply_w_ids = 8;
  repeated uint32 o_ol_quantities = 9;
  required uint32 o_entry_d = 10;
  required bool all_local = 11;
  required bool sequential = 12;
  repeated uint32 est_tables = 13;
}

message OrderStatus {
  required uint32 c_w_id = 1;
  required uint32 c_d_id = 2;
  required uint32 c_id = 3;
  required uint32 o_id = 4;
  required bool c_by_last_name = 5;
  required string c_last = 6;
}

message Payment {
  required uint32 w_id = 1;
  required uint32 d_id = 2;
  required uint32 d_w_id = 3;
  required uint32 c_w_id = 4;
  required uint32 c_d_id = 5;
  required uint32 c_id = 6;
  required uint32 h_amount = 7;
  required uint32 h_date = 8;
  required uint32 random_row_id = 9;
  required bool c_by_last_name = 10;
  required string c_last = 11;
  required bool sequential = 12;
  repeated uint32 est_tables = 13;
}

message StockLevel {
  required uint32 w_id = 1;
  required uint32 d_id = 2;
  required uint32 min_quantity = 3;
}

message PolicyChange {
  required uint32 w_id = 1;
}